### ISSUE TITLE
Odd error when URI-hacking replay

### ISSUE THREAD

**Issue by machawk1** [oduwsdl/ipwb#667]
Latest ipwb master, 850389f

After starting replay, go to localhost:5000/lorem where `lorem` is any random string. The result is a 404:

![Screen Shot 2020-06-23 at 11 10 43 AM](https://user-images.githubusercontent.com/2514780/85421384-5cf12d00-b542-11ea-8dac-0081b056b228.png)

The message is a little odd. `None` is likely surfaced up from the Python code and does not quite make sense in this context. Clicking on the Link literal produces an equally confusing 404 message.

![Screen Shot 2020-06-23 at 11 10 46 AM](https://user-images.githubusercontent.com/2514780/85421520-8316cd00-b542-11ea-9468-b84acf6e5733.png)

The 404 page could provide more information on what went wrong and correctly parse out (presumably the date or lack-thereof) what is provided for the message body.

**Comment by ibnesayeed** [comment]
This is coming form an old route that we discussed earlier and pushed back to revisit its removal later (see the TODO note).

https://github.com/oduwsdl/ipwb/blob/850389f0a15b5cdd521d24c9f9cf54a2dfa478a4/ipwb/replay.py#L607-L609

**Comment by machawk1** [comment]
Removing the route causes the same URI to simple return "Error". The decorator can probably be removed though we may want to handle URIs like it better. We should also run the test suite to see if it surfaces up any functionality that relies on this route. Is there a GitHub issue associated with its removal?

**Comment by ibnesayeed** [comment]
> Removing the route causes the same URI to simple return "Error".

Which is the expected behavior. It will be a generic 404, albeit, not looking for a memento. We can have a custom 404 page for generic ones if we want. Alternatively, we can define a catch-all route towards the end for situations like these.

> Is there a GitHub issue associated with its removal?

We discussed a related matter in #342 and #450.

**Comment by machawk1** [comment]
Right, #342 was the one I recall discussing. Current behavior with "Error" is an HTTP 500 (via curl), oddly, as the reported exception is `<class 'werkzeug.exceptions.NotFound'>, <NotFound '404: Not Found'>`. This is attributed to ipwb replay not having a default 404 handler, e.g.,

```
@app.errorhandler(404)
def page_not_found(e):
    return "Not found", 404
```

Adding the handler _might_ be a good alternative to a catch-all route. What do you think, @ibnesayeed?

**Comment by ibnesayeed** [comment]
Try adding that handler, make sure it does not overwrite existing memento errors. I mean generic error should only be returned for non-specific resources.

**Comment by machawk1** [comment]
With the handler in-place and the route decorator removed, I can still access the original 404 page for URI-Ms that have no URI-R in the index.

![Screen Shot 2020-06-23 at 4 02 08 PM](https://user-images.githubusercontent.com/2514780/85454475-fb918400-b56a-11ea-9095-9b303b4e33d4.png)

![Screen Shot 2020-06-23 at 4 02 29 PM](https://user-images.githubusercontent.com/2514780/85454501-ffbda180-b56a-11ea-87b2-bc3d371fa114.png)

I would anticipate making the two have a consistent style, even if showing different information, but that is a different GitHub issue.

**Comment by ibnesayeed** [comment]
This looks good. However, I would also suggest adding a "recorded 404 memento" in tests and inspect if necessary headers are returned as expected.
